File: apps/web/src/shared/api/mappers.ts
import type { PublicUserResponse, SkillResponse } from './types';

type SkillApiResponse = {
  id: string;
  name: string;
  created_at: string;
  updated_at: string;
};

type PublicUserApiResponse = {
  id: string;
  name: string;
  city: string;
  skills: SkillApiResponse[];
  created_at: string;
};

export function mapSkill(raw: SkillApiResponse): SkillResponse {
  return {
    id: raw.id,
    name: raw.name,
    createdAt: raw.created_at,
    updatedAt: raw.updated_at,
  };
}

export function mapPublicUser(raw: PublicUserApiResponse): PublicUserResponse {
  return {
    id: raw.id,
    name: raw.name,
    city: raw.city,
    skills: raw.skills.map(mapSkill),
    createdAt: raw.created_at,
  };
}

export type { SkillApiResponse, PublicUserApiResponse };

File: apps/api/internal/domain/service/mappers.go
package service

import (
	"juntae-api/internal/domain/dto"
	"juntae-api/internal/domain/model"
)

func mapPublicUserResponse(u *model.User) dto.PublicUserResponse {
	skills := make([]dto.SkillResponse, len(u.Skills))
	for i := range u.Skills {
		skills[i] = mapSkillResponse(&u.Skills[i])
	}
	return dto.PublicUserResponse{
		ID:        u.ID,
		Name:      u.Name,
		City:      u.City,
		Skills:    skills,
		CreatedAt: u.CreatedAt,
	}
}
